### PR TITLE
DRY up ignore related conditionals

### PR DESCRIPTION
**Pull request by rodaine**

### CODE REVIEW

File: internal/evaluator/builder.go [hunk 1]
```diff
@@ -266,21 +260,10 @@ func (bldr *Builder) processIgnoreEmpty(
...
-	val.IgnoreEmpty = forItems && (constraints.GetIgnoreEmpty() ||
-		constraints.GetIgnore() == validate.Ignore_IGNORE_IF_UNPOPULATED ||
-		constraints.GetIgnore() == validate.Ignore_IGNORE_IF_DEFAULT_VALUE)
-	switch {
-	case !val.IgnoreEmpty:
-		// only need the zero value for checking ignore_empty constraint
+	val.IgnoreEmpty = forItems && bldr.shouldIgnoreEmpty(constraints)
+	if val.IgnoreEmpty {
+		val.Zero = bldr.zeroValue(fdesc, forItems)
 		return nil
```
Comment (pkwarren) [inline internal/evaluator/builder.go:266]: Can remove this return and fall back to one immediately after.

File: internal/evaluator/builder.go [hunk 2]
```diff
@@ -207,18 +207,12 @@ func (bldr *Builder) buildField(
 		Descriptor: fieldDescriptor,
 		Required:   fieldConstraints.GetRequired(),
 		IgnoreEmpty: fieldDescriptor.HasPresence() ||
-			fieldConstraints.GetIgnoreEmpty() ||
-			fieldConstraints.GetIgnore() == validate.Ignore_IGNORE_IF_UNPOPULATED ||
-			fieldConstraints.GetIgnore() == validate.Ignore_IGNORE_IF_DEFAULT_VALUE,
-		IgnoreDefault: fieldDescriptor.HasPresence() && fieldConstraints.GetIgnore() == validate.Ignore_IGNORE_IF_DEFAULT_VALUE,
+			bldr.shouldIgnoreEmpty(fieldConstraints),
```
Comment (pkwarren) [inline internal/evaluator/builder.go:210]: nit: Could put these back on one line if you want now that they're much shorter.

**Review by pkwarren — APPROVED** [review]

File: internal/evaluator/builder.go (hunk 1 above)
Comment (rodaine) [reply internal/evaluator/builder.go:266]: weird that a linter didn't catch that...

File: internal/evaluator/builder.go (hunk 2 above)
Comment (rodaine) [reply internal/evaluator/builder.go:210]: kinda like it on multiple lines personally 😅

**Review by pkwarren — APPROVED** [review]

**Pull request merged**